index.js

```diff
@@ -30,19 +30,29 @@ module.exports = function(options) {
     var parsedBody = getParsedBody(this);
 
     var opt = {
-      url: url + '?' + this.querystring,
+      url: url + (this.querystring ? '?' + this.querystring : ''),
       headers: this.header,
       encoding: null,
       method: this.method,
       body: parsedBody
     };
+
     // set 'Host' header to options.host (without protocol prefix), strip trailing slash
     if (options.host) opt.headers.host = options.host.slice(options.host.indexOf('://')+3).replace(/\/$/,'');
 
     if (options.requestOptions) {
-      Object.keys(options.requestOptions).forEach(function (option) { opt[option] = options.requestOptions[option]; });
-    }
+      var ctx = this;
 
+      Object.keys(options.requestOptions).forEach(function (optionName) {
```

Comment:
I pushed support for this in latest commit. Please check it out and let me know if further adjustments are required.